packages/amplify-cli-core/src/amplify-node-pkg-detector/parser-factory.ts

```diff
@@ -26,7 +26,7 @@ export class LockFileParserFactory {
       }
       default:
         throw new AmplifyError('UnsupportedLockFileTypeError', {
-          message: 'Unsupported lockfile type ' + `${packageManager.lockFile} provided. Only 'npm' or 'yarn' is currently ` + 'supported.',
+          message: `Unsupported lockfile type ${packageManager.lockFile} provided. Only 'npm', 'yarn', and 'pnpm' are currently supported.`,
```

Comment:
In this file above we have `LockFileType` which current has Yarn or NPM, do we not need PNPM here as well? and Yarn 3?

Reply:
this parser-factory logic ends up being used only here, for a migration message: https://github.com/lazpavel/amplify-cli/blob/lambda-package-manager/packages/amplify-provider-awscloudformation/src/print-cdk-migration-warning.ts#L106-L133

since PNPM is newly added we don't currently need it being added here